ufcg/src/test/java/br/com/askufcg/services/user/UserServiceTest.java

```diff
@@ -0,0 +1,158 @@
+package br.com.askufcg.services.user;
+
+import br.com.askufcg.exceptions.BadRequestException;
+import br.com.askufcg.exceptions.NoContentException;
+import br.com.askufcg.exceptions.NotFoundException;
+import br.com.askufcg.models.User;
+import br.com.askufcg.repositories.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class UserServiceTest {
+    @InjectMocks
+    private UserService userService = new UserServiceImpl();
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    private List<User> mockedUsers;
+    private List<User> emptyList;
+    private User mockedUser;
+    private User mockedUser2;
+
+    @BeforeEach
+    void setUp() {
+        this.mockedUsers = new ArrayList<>();
+        this.emptyList = new ArrayList<>();
+
+        this.mockedUser = new User(14L, "wellisson@mail.com", "hashed", "Wellisson", "Gomes", "", "", "");
+        this.mockedUser2 = new User(2L, "cicera@mail.com", "hashed", "Cicera", "Gomes","sem avatar", "linkedin", "github");
+
+        this.mockedUsers.add(new User(1L, "wellisson@mail.com", "hashed", "Wellisson", "Gomes", "", "", ""));
+        this.mockedUsers.add(new User(2L, "lucas@mail.com", "hashed", "Lucas", "Gomes", "", "", ""));
+
+        when(bCryptPasswordEncoder.encode(Mockito.any())).thenReturn(mockedUser.getPassword());
```

Comment:
Achei massa, o que isso faz?

Reply:
O mockito faz um mock do objeto `BCryptPasswordEncoder` para que possamos manipular o retorno. Como ele gera um hash e não temos como saber qual será (para comparar), então forçamos ele a ser um hash/string qualquer.